lib/pf/api.pm

```diff
@@ -439,6 +451,130 @@ sub node_information : Public {
     return $node_info;
 }
 
+sub notify_configfile_changed : Public {
+    my ($class, %postdata) = @_;
+    my $logger = pf::log::get_logger;
+    my @require = qw(server conf_file);
+    my @found = grep {exists $postdata{$_}} @require;
+    return unless validate_argv(\@require, \@found);
+
+    # we light expire pfconfig cluster configuration on this server so it uses the distributed configuration
+    my $payload = {
+        method => "expire",
+        namespace => 'config::Cluster',
+        light => 1,
+    };
+    pfconfig::util::fetch_decode_socket(encode_json($payload));
+
+    my $master_server = $ConfigCluster{$postdata{server}};
+    die "Master server is not in configuration" unless ($master_server);
+
+    my $apiclient = pf::api::jsonrpcclient->new(proto => 'https', host => $master_server->{management_ip});
+
+    eval {
+        my %data = ( conf_file => $postdata{conf_file} );
+        my ($result) = $apiclient->call( 'download_configfile', %data );
+        open(my $fh, '>', $postdata{conf_file}) or die "Cannot open file $postdata{conf_file} for writing. This is excessively bad. Run '/usr/local/pf/bin/pfcmd fixpermissions'";
+        print $fh $result;
+        close($fh);
+        use pf::config::cached;
+        pf::config::cached::updateCacheControl();
+        pf::config::cached::ReloadConfigs(1);
+
+        $logger->info("Successfully downloaded configuration $postdata{conf_file} from $postdata{server}");
+    };
+    if($@){
+        $logger->error("Couldn't download configuration file $postdata{conf_file} from $postdata{server}. $@");
+    }
+
+    return 1;
+}
+
+sub download_configfile : Public {
+    my ($class, %postdata) = @_;
+    my @require = qw(conf_file);
+    my @found = grep {exists $postdata{$_}} @require;
+    return unless validate_argv(\@require, \@found);
+
+    use File::Slurp;
+    die "Config file $postdata{conf_file} doesn't exist" unless(-e $postdata{conf_file});
+    my $config = read_file($postdata{conf_file});
+
+    return $config;
+}
+
+sub expire_cluster : Public {
+    my ($class, %postdata) = @_;
+    my @require = qw(namespace conf_file);
+    my @found = grep {exists $postdata{$_}} @require;
+    return unless validate_argv(\@require, \@found);
+
+    my $logger = pf::log::get_logger;
+
+    $postdata{light} = 0;
+    expire($class, %postdata);
+
+    foreach my $server (@cluster_servers){
+        next if($host_id eq $server->{host});
+        my $apiclient = pf::api::jsonrpcclient->new(proto => 'https', host => $server->{management_ip});
+        my %data = (
+            namespace => $postdata{namespace},
+            light => 1
+        );
+        eval {
+            $apiclient->call('expire', %data ); 
+        };
+
+        if($@){
+            $logger->error("Can't connect to $server->{management_ip} to expire the configuration. $@")
```

Comment:
That logging message was bad. This means an error occured during the request so the next step may work if the issue is not network related. I'll fix the logging